### PR TITLE
Rebase user-model/main into main

### PR DESCRIPTION
**Pull request by jinliu9508**
This PR rebase user-model/main into main.
Note that user-model/main-rebased is a rebased version of user-model/main.

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/702)

### CODE REVIEW

**Review by shepherd-l — APPROVED** [review]
Lgtm!

File: OneSignalExample/Packages/packages-lock.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "depth": 0,
       "source": "local",
       "dependencies": {
-        "com.onesignal.unity.core": "3.0.11"
+        "com.onesignal.unity.core": "5.1.5"
```
Comment (shepherd-l) [inline OneSignalExample/Packages/packages-lock.json:8]: I think this should be "5.1.0"

**Review by shepherd-l — CHANGES_REQUESTED** [review]

File: OneSignalExample/Assets/OneSignal/CHANGELOG.md [hunk 2]
```diff
@@ -5,9 +5,95 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-## [3.0.11]
```
Comment (shepherd-l) [inline OneSignalExample/Assets/OneSignal/CHANGELOG.md:8]: I think this should be included from the main branch

**Review by shepherd-l — CHANGES_REQUESTED** [review]

File: OneSignalExample/Assets/OneSignal/CHANGELOG.md (hunk 2 above)
Comment (jinliu9508) [reply OneSignalExample/Assets/OneSignal/CHANGELOG.md:8]: Ok I just added it back there. Probably got deleted using the auto resolver.

File: OneSignalExample/Assets/OneSignal/CHANGELOG.md [hunk 3]
```diff
@@ -5,23 +5,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-## [5.1.0]
```
Comment (shepherd-l) [inline OneSignalExample/Assets/OneSignal/CHANGELOG.md]: I think the 5.1.0 change log notes should be included.

File: OneSignalExample/Assets/OneSignal/CHANGELOG.md [hunk 4]
```diff
@@ -196,9 +277,16 @@ If you run into any problems, please don’t hesitate to [open an issue](https:/
   - If you are updating from a previous version of the OneSignal Unity SDK please follow the Unity Asset Store instructions in
       the [README](https://github.com/OneSignal/OneSignal-Unity-SDK/README.md#unity-asset-store) to ensure a smooth transition.
 
-[Unreleased]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/3.0.11...HEAD
```
Comment (shepherd-l) [inline OneSignalExample/Assets/OneSignal/CHANGELOG.md:199]: I think the order should go from to the latest player model version (3.0.11) and then switch to the user model beta versions and up.

Like this:
```
...
[5.0.1]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/5.0.0...5.0.1
[5.0.0]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/5.0.0-beta.3...5.0.0
[5.0.0-beta.3]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/5.0.0-beta.2...5.0.0-beta.3
[5.0.0-beta.2]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/5.0.0-beta.1...5.0.0-beta.2
[5.0.0-beta.1]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/3.0.11...5.0.0-beta.1
[3.0.11]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/3.0.10...3.0.11
...
```

And to copy what was done in the past where even though the main version was released after the beta release, it is still ordered from the latest main release to the beta release of the next version.

2.14.6 was released after 3.0.0 beta1 and beta2
```
[3.0.0-beta.2]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/3.0.0-beta.1...3.0.0-beta.2
[3.0.0-beta.1]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/2.14.6...3.0.0-beta.1
[2.14.6]: https://github.com/OneSignal/OneSignal-Unity-SDK/compare/2.14.5...2.14.6
```
What do you think?

**Review by shepherd-l — CHANGES_REQUESTED** [review]

**Review by shepherd-l — APPROVED** [review]
Lgtm!
Tested OneSignal initialization on an emulated Pixel 4 with Android 12 and physical iPhone 12 with iOS 15.5 built with Unity 2022.3.10f1 with the OneSignal example app

**Pull request merged**